feature_engine/prediction/target_mean_prediction.py

```diff
@@ -0,0 +1,271 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator
+from sklearn.metrics import accuracy_score, r2_score
+from sklearn.pipeline import Pipeline
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+from feature_engine.discretisation import (
+    EqualFrequencyDiscretiser,
+    EqualWidthDiscretiser
+)
+from feature_engine.encoding import MeanEncoder
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_all_variables,
+)
+
+
+class TargetMeanPredictor(BaseEstimator):
+    """
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of input variables. If None, the estimator will evaluate will use all
+        variables as input fetures.
+
+    bins: int, default=5
+        If the dataset contains numerical variables, the number of bins into which
+        the values will be sorted.
+
+    strategy: str, default='equal_width'
+        Whether the bins should of equal width ('equal_width') or equal frequency
+        ('equal_frequency').
+
+    Attributes
+    ----------
+
+
+    Methods
+    -------
+    fit:
+
+    predict:
+
+    Notes
+    -----
+
+
+    See Also
+    --------
+    feature_engine.encoding.MeanEncoder
+    feature_engine.discretisation.EqualWidthDiscretiser
+    feature_engine.discretisation.EqualFrequencyDiscretiser
+
+    References
+    ----------
+
+
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        bins: int = 5,
+        strategy: str = "equal-width",
```

Comment:
I believe we use underscore instead of hyphen in the strategy arguments. Can you check other classes and change the hyphen for underscore? so equal**_**width instead of equal-width?